.azure-pipelines/steps/macos.yml

```diff
@@ -0,0 +1,47 @@
+steps:
+- template: show-disk-usage.yml
+
+- bash: |
+    export PATH=$PATH:$HOME/.local/bin:$HOME/Library/Python/2.7/bin/
+    mkdir -p $HOME/rustsrc
+    echo "##vso[task.setvariable variable=PATH;]$PATH"
+
+    curl -fo /usr/local/bin/sccache https://s3-us-west-1.amazonaws.com/rust-lang-ci2/rust-ci-mirror/2018-04-02-sccache-x86_64-apple-darwin
+    chmod +x /usr/local/bin/sccache
+
+    curl -fo /usr/local/bin/stamp https://s3-us-west-1.amazonaws.com/rust-lang-ci2/rust-ci-mirror/2017-03-17-stamp-x86_64-apple-darwin
+    chmod +x /usr/local/bin/stamp
+
+    export CC=`pwd`/clang+llvm-7.0.0-x86_64-apple-darwin/bin/clang
+    echo "##vso[task.setvariable variable=CC]$CC"
+
+    export CXX=`pwd`/clang+llvm-7.0.0-x86_64-apple-darwin/bin/clang++
+    echo "##vso[task.setvariable variable=CXX]$CXX"
+
+    echo "##vso[task.setvariable variable=AR]ar"
+  displayName: Prep
+
+- bash: brew install gnu-tar
+  displayName: install a tar that works well
+
+- bash: |
+    curl -f http://releases.llvm.org/7.0.0/clang+llvm-7.0.0-x86_64-apple-darwin.tar.xz | tar xJf -
+  displayName: Download clang
+
+- bash: |
+    brew update
+    brew install xz
+    brew install swig
+  condition: and(succeeded(), eq(variables['RUST_CHECK_TARGET'],'dist'))
+  displayName: Install xz and swigw
+
+- bash: |
+    export RUN_SCRIPT="$BUILD_SOURCESDIRECTORY/src/ci/init_repo.sh . $HOME/rustsrc && src/ci/run.sh"
+    echo "##vso[task.setvariable variable=RUN_SCRIPT]$RUN_SCRIPT"
```

Comment:
FWIW this "RUN_SCRIPT" business I feel like was largely workarounds for travis or just getting things to work over time, there's probably better ways to organize this nowadays, but we can always figure that out later!